lib/src/util/component_usage_migrator.dart

```diff
@@ -0,0 +1,791 @@
+  }
+
+  void yieldPropManualMigratePatch(PropAssignment prop) {
+    yieldPropFixmePatch(prop, 'manually migrate');
+  }
+
+  /// Yields a patch with a fix-me comment before a given [prop]
+  /// with a custom [message].
+  ///
+  /// Inserts extra newlines as necessary to avoid the comment getting "stuck"
+  /// to the previous line, or adding blank lines.
+  ///
+  /// For example:
+  /// ```dart
+  /// // Flagging "id" for this usage...
+  /// (Dom.div()..id = '')();
+  ///
+  /// // ...results in this...
+  /// (Dom.div()
+  ///   // FIXME ...
+  ///   ..id = ''
+  /// )();
+  ///
+  /// // ...and not this...
+  /// (Dom.div() // FIXME ...
+  ///   ..id = ''
+  /// )();
+  /// ```
+  void yieldPropFixmePatch(PropAssignment prop, String message) {
+    yieldBuilderMemberFixmePatch(prop, '${prop.name.name} prop - $message');
+  }
+
+  /// Yields a patch with a fix-me comment before a given [access]
+  /// with a custom [message].
+  ///
+  /// Inserts extra newlines as necessary to avoid the comment getting "stuck"
+  /// to the previous line, or adding blank lines.
+  ///
+  /// For example:
+  /// ```dart
+  /// // Flagging "id" for this usage...
+  /// (Dom.div()..id = '')();
+  ///
+  /// // ...results in this...
+  /// (Dom.div()
+  ///   // FIXME ...
+  ///   ..id = ''
+  /// )();
+  ///
+  /// // ...and not this...
+  /// (Dom.div() // FIXME ...
+  ///   ..id = ''
+  /// )();
+  void yieldBuilderMemberFixmePatch(
+      BuilderMemberAccess access, String message) {
+    // Add an extra newline beforehand so that the comment doesn't end up on
+    // the same line as the cascade target (the builder) in single-prop cascades.
+    // Add a space so that dartfmt indents comment with the next line as opposed to
+    // keeping it at the beginning of the line.
+    // This formatting makes the output nicer, but more importantly it keeps
+    // formatting of expected output in tests more consistent and easier to predict.
+    final needsLeadingNewline = access.parentCascade != null &&
+        context.sourceFile.getLine(access.parentCascade!.target.end) ==
+            context.sourceFile.getLine(access.node.offset);
+    yieldInsertionPatch(
+        (needsLeadingNewline ? '\n ' : '') +
+            lineComment('$fixmePrefix - $message'),
+        access.node.offset);
+  }
+
+  /// Yields a patch with a fix-me comment before a given [child]
+  /// with a custom [message].
+  ///
+  /// Inserts extra newlines as necessary to avoid the comment getting "stuck"
+  /// to the previous line, or adding blank lines.
+  ///
+  /// For example:
+  /// ```dart
+  /// // Flagging the single child for this usage...
+  /// Dom.div()('child');
+  ///
+  /// // ...results in this...
+  /// Dom.div()(
+  ///   // FIXME
+  ///   'child'
+  /// );
+  ///
+  /// // ...and not this...
+  /// Dom.div()( // FIXME ...
+  ///   'child'
+  /// );
+  void yieldChildFixmePatch(ComponentChild child, String message) {
```

Comment:
Example codeblock isn't terminated
```suggestion
  /// For example:
  /// ```dart
  /// // Flagging the single child for this usage...
  /// Dom.div()('child');
  ///
  /// // ...results in this...
  /// Dom.div()(
  ///   // FIXME
  ///   'child'
  /// );
  ///
  /// // ...and not this...
  /// Dom.div()( // FIXME ...
  ///   'child'
  /// );
  /// ```
  void yieldChildFixmePatch(ComponentChild child, String message) {
```